### PR TITLE
CMR-6745: fix errors thrown with ORed granule counts

### PR DESCRIPTION
**Pull request by mschmele**

### CODE REVIEW

File: search-app/src/cmr/search/services/query_execution/granule_counts_results_feature.clj [hunk 1]
```diff
@@ -83,24 +83,26 @@
...
+          :and))))
 
 (defn- combine-spatial-and-non-spatial-conditions
   "Helper function. If spatial conitions are ORed, combine them with the rest of
   the conditions appropriately."
   [spatial-conditions temporal-conditions collection-ids]
   (let [collection-id-conditions (q/string-conditions :collection-concept-id collection-ids true)
         ored-spatial-conditions (gc/or-conds spatial-conditions)]
-    (if (seq collection-ids)
+    (if (seq? collection-ids)
```
Comment (ygliuvt) [inline search-app/src/cmr/search/services/query_execution/granule_counts_results_feature.clj:105]: Why do we make this change?

File: system-int-test/test/cmr/system_int_test/search/granule_counts_search_test.clj [hunk 2]
```diff
@@ -311,6 +319,80 @@
...
+    (index/wait-until-indexed)
+    
+    (testing "ORed granule counts special case"
+      (let [coll6745-id (:concept-id coll6745)
+            refs (search/find-refs :collection {:include-granule-counts true
+                                                :concept-id coll6745-id
+                                                :polygon ["-21.33069,80.92296,-24.68258,80.58223,-26.80316,80.36716,-29.06056,79.60629,-24.34055,78.86559,-17.56837,79.10088,-14.62691,79.83818,-14.83213,80.79254,-21.33069,80.92296"
+                                                          "136.75804,-34.82121,135.13466,-34.7865,134.43289,-35.0361,133.97631,-35.53991,134.6189,-36.7958,136.56358,-36.90405,137.07088,-36.15672,136.75804,-34.82121"]
+                                                "options[spatial][or]" "true"})]
+        (gran-counts/granule-counts-match? :xml {coll6745 3} refs)
```
Comment (ygliuvt) [inline system-int-test/test/cmr/system_int_test/search/granule_counts_search_test.clj:380]: missing `is` in the check. Without the `is` around this line, we are not checking the granule counts.

File: system-int-test/test/cmr/system_int_test/search/granule_counts_search_test.clj [hunk 3]
```diff
@@ -311,6 +319,80 @@
...
+        (gran-counts/granule-counts-match? :xml {coll6745 3} refs)))
```
Comment (ygliuvt) [inline system-int-test/test/cmr/system_int_test/search/granule_counts_search_test.clj:381]: This line is a duplicate of the line above.

File: system-int-test/test/cmr/system_int_test/search/granule_counts_search_test.clj [hunk 4]
```diff
@@ -311,6 +319,80 @@
...
+    (testing "EDSC Pageload error case"
+      (is (= [{:id "C1200000006-PROV1"
```
Comment (ygliuvt) [inline system-int-test/test/cmr/system_int_test/search/granule_counts_search_test.clj:383]: We should avoid hardcoded identifiers in tests. Why don't we use the `granule-counts-match?` check here?

File: system-int-test/test/cmr/system_int_test/search/granule_counts_search_test.clj [hunk 5]
```diff
@@ -311,6 +319,80 @@
...
+
+    (index/wait-until-indexed)
+    
+    ;; Refresh the aggregate cache so that it includes all the granules that were added.
+    (index/full-refresh-collection-granule-aggregate-cache)
+    ;; Reindex all the collections to get the latest information.
+    (ingest/reindex-all-collections)
+    (index/wait-until-indexed)
+    
+    (testing "ORed granule counts special case"
```
Comment (ygliuvt) [inline system-int-test/test/cmr/system_int_test/search/granule_counts_search_test.clj:373]: We need to add a test where without `options[spatial][or]=true`, the search finds `{coll6745 0}`.

**Review by ygliuvt — CHANGES_REQUESTED** [review]
Need to fix tests.

File: system-int-test/test/cmr/system_int_test/search/granule_counts_search_test.clj (hunk 4 above)
Comment (mschmele) [reply system-int-test/test/cmr/system_int_test/search/granule_counts_search_test.clj:383]: `refs-match?` was throwing errors - turns out that the function doesn't dissoc the `:has-granules?` key and it was causing the check to fail

File: search-app/src/cmr/search/services/query_execution/granule_counts_results_feature.clj (hunk 1 above)
Comment (mschmele) [reply search-app/src/cmr/search/services/query_execution/granule_counts_results_feature.clj:105]: I was testing something and forgot to reverse it. `seq` vs `seq?` doesn't seem to make a difference in this case, but I'm still going to change it back

File: search-app/src/cmr/search/services/query_execution/granule_counts_results_feature.clj [hunk 6]
```diff
@@ -111,9 +113,9 @@
   "Extracts a query to find the number of granules per collection in the results from a collection query
   coll-query - The collection query
   results - the results of the collection query"
-  [coll-query results]
+  [context coll-query results]
   (let [collection-ids (query-results->concept-ids results)
-        spatial-operator (extract-spatial-operator coll-query)
+        spatial-operator (extract-spatial-operator context)
         spatial-temp-conds (->> coll-query
```
Comment (ygliuvt) [inline search-app/src/cmr/search/services/query_execution/granule_counts_results_feature.clj:119]: It is confusing that we extract spatial and temporal conditions out of the query, then extract spatial and temporal conditions separately from the `spatial-temp-conds`. Why don't we just extract spatial conditions and temporal conditions separately from the query and construct the final condition based on the spatial operator?

File: system-int-test/test/cmr/system_int_test/search/granule_counts_search_test.clj [hunk 7]
```diff
@@ -311,6 +319,82 @@
...
+      (let [coll6745-id (:concept-id coll6745)
+            refs (search/find-refs :collection {:include-granule-counts true
+                                                :concept-id coll6745-id
+                                                :polygon ["-21.33069,80.92296,-24.68258,80.58223,-26.80316,80.36716,-29.06056,79.60629,-24.34055,78.86559,-17.56837,79.10088,-14.62691,79.83818,-14.83213,80.79254,-21.33069,80.92296"
```
Comment (ygliuvt) [inline system-int-test/test/cmr/system_int_test/search/granule_counts_search_test.clj:377]: make the polygon value a variable and used it in both test cases to make it clear that the only thing changed between the test cases is the `options[spatial][or]` parameter.

File: system-int-test/test/cmr/system_int_test/search/granule_counts_search_test.clj [hunk 8]
```diff
@@ -311,6 +319,82 @@
...
+                                                          "136.75804,-34.82121,135.13466,-34.7865,134.43289,-35.0361,133.97631,-35.53991,134.6189,-36.7958,136.56358,-36.90405,137.07088,-36.15672,136.75804,-34.82121"]
+                                                "options[spatial][or]" "true"})]
+        (is (gran-counts/granule-counts-match? :xml {coll6745 3} refs))))
+    (testing "ANDed granule counts special case"
```
Comment (ygliuvt) [inline system-int-test/test/cmr/system_int_test/search/granule_counts_search_test.clj:381]: nit: add empty lines between test cases to make it more readable.

File: system-int-test/test/cmr/system_int_test/search/granule_counts_search_test.clj [hunk 9]
```diff
@@ -311,6 +319,82 @@
...
+                                                          "136.75804,-34.82121,135.13466,-34.7865,134.43289,-35.0361,133.97631,-35.53991,134.6189,-36.7958,136.56358,-36.90405,137.07088,-36.15672,136.75804,-34.82121"]
+                                                "options[spatial][or]" "false"})]
+        (is (gran-counts/granule-counts-match? :xml {coll6745 0} refs))))
+    (testing "EDSC Pageload error case"
+      (let [refs (search/find-refs :collection {:include-granule-counts true
+                                                :has-granules-or-cwic true
+                                                "options[science_keywords_h][or]" "true"
+                                                "options[spatial][or]" "true"
+                                                "options[temporal][limit_to_granules]" "true"})]
+        (is (d/refs-match? [coll6745] refs))
```
Comment (ygliuvt) [inline system-int-test/test/cmr/system_int_test/search/granule_counts_search_test.clj:395]: The reference check is in the `granule-counts-match?` check in the line below already. We can remove this line.

**Review by ygliuvt — APPROVED** [review]

**Review by daniel-zamora — APPROVED** [review]

**Pull request merged**